src/compas_fab/backends/kinematics/analytical_inverse_kinematics.py

```diff
@@ -0,0 +1,143 @@
+    each other. That is why for certain use cases, e.g. custom cartesian
+    path planning it makes sense to keep the sorting and set the ones
+    that are out of joint limits or in collison to `None`.
+
+    Examples
+    --------
+
+    >>> ik_solver = Analytical_InverseKinematics()
+    """
+
+    def __init__(self, client=None):
+        self.client = client
+
+    def inverse_kinematics(self, robot, frame_RCF, start_configuration=None, group=None, options=None):
+
+        solutions = self._inverse_kinematics(frame_RCF)
+
+        configurations = self.joint_angles_to_configuration(robot, solutions)
+
+        # check collisions for all configurations (sets those to `None` that are not working)
+        if options and "check_collision" in options and options["check_collision"] is True:
+            for i, config in enumerate(configurations):
+                try:
+                    self.client.check_collisions(robot, config)
+                except BackendError:
+                    configurations[i] = None
+
+        # fit configurations within joint bounds (sets those to `None` that are not working)
+        configurations = self.try_to_fit_configurations_between_bounds(robot, configurations)
+
+        # removes the `None` ones
+        if options and "cull" in options and options["cull"] is True:
+            configurations = [c for c in configurations if c is not None]
+
+        return configurations
+
+    def _inverse_kinematics(self, frame):
+        pass
+
+    def joint_angles_to_configurations(self, robot, solutions):
+        joint_names = robot.get_configurable_joint_names()
+        return [Configuration.from_revolute_values(q, joint_names=joint_names) for q in solutions]
+
+    def try_to_fit_configurations_between_bounds(self, robot, configurations):
+        """
+        """
+        j1, j2, j3, j4, j5, j6 = robot.get_configurable_joints()
+        for i, c in enumerate(configurations):
+            if c is None:
+                continue
+            a1, a2, a3, a4, a5, a6 = c.values()
+            try:
+                a1 = fit_within_bounds(a1, j1.limit.lower, j1.limit.upper)
+                a2 = fit_within_bounds(a2, j2.limit.lower, j2.limit.upper)
+                a3 = fit_within_bounds(a3, j3.limit.lower, j3.limit.upper)
+                a4 = fit_within_bounds(a4, j4.limit.lower, j4.limit.upper)
+                a5 = fit_within_bounds(a5, j5.limit.lower, j5.limit.upper)
+                a6 = fit_within_bounds(a6, j6.limit.lower, j6.limit.upper)
+                configurations[i].joint_values = [a1, a2, a3, a4, a5, a6]
+            except AssertionError:
+                configurations[i] = None
+        return configurations
+
+
+class UR5_Analytical_IK(Analytical_InverseKinematics):
+
+    def _inverse_kinematics(self, frame):
+        from compas_fab.backends.kinematics.offset_wrist_kinematics import UR5
+        return UR5().inverse(frame)
```

Comment:
It's not clear to me where, or if at all, is the link between a robot model and its specific analytical ik solver instance. Can we somehow determine automatically or with some bit of extra data how to instantiate the correct ik solver?

Reply:
do you mean that the IK solver needs to correspond to the `robot.model.name` and that we can then call it via `eval`?